src/routes/accounts.py

```
@@ -15,11 +17,151 @@
+            ActivationTokenModel.token == user_data.token
+        ).where(ActivationTokenModel.user.has(email == user_data.email)))
+
+    token_obj = result.scalars().firts()
+
+    expires_at_utc= cast(datetime, token_obj.expires_at).replace(tzinfo=timezone.utc)
+    now_utc = cast(datetime, datetime.utcnow().replace(tzinfo=timezone.utc))
+
+    if not token_obj or expires_at_utc < now_utc:
+        raise HTTPException(status_code=400, detail="Invalid or expired activation token.")
+
+    user = token_obj.user
+    if user.is_active:
+        raise HTTPException(status_code=400, detail="User account is already active.")
+
+    user.is_active = True
+    await db.delete(token_obj)
+    await db.commit()
+    await db.refresh(user)
+    return {"message": "User account activated successfully."}
+
+
+@router.post("/login/",
+             response_model=UserLoginResponseSchema,
+             status_code=status.HTTP_200_OK
+             )
+async def login_user(
+        login_data: UserLoginRequestSchema,
+        db: AsyncSession = Depends(get_db),
+        jwt_manager: JWTAuthManagerInterface = Depends(get_jwt_auth_manager),
+        settings: BaseAppSettings = Depends(get_settings)
+        ):
+    result = await db.execute(
+        select(UserModel).where(
+            UserModel.email == login_data.email))
+    user = result.scalars().first()
+    if not user or not user.verify_password(login_data.password):
+        raise HTTPException(status_code=401, detail="Invalid email or password.")
+    if not user.is_active:
+        raise HTTPException(status_code=403, detail="User account is not activated.")
+
+    access_token = jwt_manager.create_access_token({"sub": str(user.id)})
+    refresh_token = jwt_manager.create_refresh_token({"sub": str(user.id)})
+
+    try:
+        new_refresh_token = RefreshTokenModel(user=user.id, token=refresh_token)
+        db.add(new_refresh_token)
+        await db.commit()
+
+    except:
```

Avoid bare except: `except:` will catch everything and mask errors. Use `except SQLAlchemyError:` when catching DB-related exceptions so you rollback and re-raise the correct HTTPException while preserving other unexpected exceptions.